make/jdk/src/classes/build/tools/generatezip/GenerateZip.java

```diff
@@ -0,0 +1,299 @@
+            Path   filepath = entry.getValue();
+
+            File f = filepath.toFile();
+            entries.add(f);
+        }
+    }
+
+    // Create new zip from entries
+    boolean create(OutputStream out) throws IOException
+    {
+        try (ZipOutputStream zos = new ZipOutputStream(out)) {
+            for (File file: entries) {
+                addFile(zos, file);
+            }
+        }
+        return true;
+    }
+
+    // Ensure a consistent entry name format
+    String entryName(String name) {
+        name = name.replace(File.separatorChar, '/');
+
+        if (name.startsWith("/")) {
+            name = name.substring(1);
+        } else if (name.startsWith("./")) {
+            name = name.substring(2);
+        }
+        return name;
+    }
+
+    // Add File to Zip
+    void addFile(ZipOutputStream zos, File file) throws IOException {
+        String name = file.getPath();
+        boolean isDir = file.isDirectory();
+        if (isDir) {
+            name = name.endsWith(File.separator) ? name : (name + File.separator);
+        }
+        name = entryName(name);
+
+        if (name.equals("") || name.equals(".") || name.equals(zname)) {
+            return;
+        }
+
+        long size = isDir ? 0 : file.length();
+
+        if (verbose) {
+            System.out.println("Adding: "+name);
+        }
+
+        ZipEntry e = new ZipEntry(name);
+        // If we are adding a new entry and SOURCE_DATE_EPOCH is set then use that time
+        if (sourceDateEpochMillis != -1) {
+            e.setTime(sourceDateEpochMillis);
+        } else {
+            e.setTime(file.lastModified());
+        }
+        if (size == 0) {
+            e.setMethod(ZipEntry.STORED);
+            e.setSize(0);
+            e.setCrc(0);
+        }
+        zos.putNextEntry(e);
+        if (!isDir) {
+            byte[] buf = new byte[8192];
+            int len;
+            InputStream is = new BufferedInputStream(new FileInputStream(file));
+            while ((len = is.read(buf, 0, buf.length)) != -1) {
+                zos.write(buf, 0, len);
+            }
+            is.close();
+        }
+        zos.closeEntry();
+    }
+
+    void usageError() {
+        error(
+        "Usage: GenerateZip [-v] -f <zip_file> <files_or_directories>\n" +
+        "Options:\n" +
+        "   -v  verbose output\n" +
+        "   -f  specify archive file name to create\n" +
+        "If any file is a directory then it is processed recursively.\n");
+    }
+
+    void fatalError(Exception e) {
+        e.printStackTrace();
+    }
+
+    protected void error(String s) {
+        System.err.println(s);
+    }
+
+    public static String[] parse(String[] args)
+    {
+        ArrayList<String> newArgs = new ArrayList<String>(args.length);
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+            newArgs.add(arg);
+        }
+        return newArgs.toArray(new String[newArgs.size()]);
```

Comment:
something might be missing here. It just adds args to a list and makes it an array again + If the language level allows it toArray(String[]::new) could be used.